### PR TITLE
:graphics can .setFont with system fonts.  With tests and updated documentation.

### PR DESCRIPTION
**Pull request by benzenwen**
graphics-2d had a stubbed out DefaultFontMapper, never registered with system fonts.  Using .setFont in a :graphics tag would not honor system fonts.  Additionally, finding the correct strings to refer to system fonts was difficult.  This pull request enables .setFont in a :graphics tag with the same known system font directories as the rest of clj-pdf and provides a utility to enumerate the font names and mapped nicknames.  A new test cli-pdf.test.graphics-2d is also provided that was tested on MacOS 10.12.5 and Ubuntu 16.04.2 LTS.  Documentation and the example file were also updated.  I tend to commit often, and am happy to squash commits for hygiene if desired (I'm a relative noob with GitHub pull requests, so apologies if this is not needed).  Thank you for writing and maintaining this repo.

### CODE REVIEW

File: src/clj/clj_pdf/graphics_2d.clj [hunk 1]
```diff
@@ -26,3 +32,54 @@
...
+          files (.list fd)
+          paths (map #(full-path dir %) files)]
+      (doseq [path paths]
+        (if (.isDirectory (io/file path))
+          (g2d-register-fonts-helper
+           path recursive))))))
+
+(defn g2d-register-fonts
+  "Walks common font directories and registers them for use."
+  []
```
Comment (yogthos) [inline src/clj/clj_pdf/graphics_2d.clj]: would it make sense to allow passing optional custom dirs for fonts, something like:

```clojure
[& [font-dirs]]
...
(doseq [[dir recursive] (or font-dirs common-font-dirs)]
```

**Comment by yogthos** [comment]
Thanks, the pr looks great! I had one small comment, but otherwise I'd be glad to merge it in and push out a release.

**Comment by benzenwen** [comment]
cool idea, let me take a stab at it.

**Comment by yogthos** [comment]
👍

**Comment by yogthos** [comment]
Just pushed out `2.2.28` to Clojars, and thanks again for spearheading this.

**Comment by benzenwen** [comment]
My pleasure!  Thank you for the fast turnaround.

**Pull request merged**